controllers/users/authenticate-web.js

```diff
@@ -21,68 +18,52 @@ const authenticateWeb = async (req, res) => {
   /**
    * @type {RegisterBody}
    */
-  const users = req.body;
+  const {exchangeToken} = req.body;
 
-  const checkUser = await User.findOne({
-    where: {
-      human_id: users.human_id
-    }
-  });
+  const verifyTokenResponse = await HumanIdService.verifyExchangeToken(exchangeToken);
+  if (!verifyTokenResponse.success) {
+    return res.status(500).json({
+      status: 'error on verify exchange token',
+      code: 500
+    });
+  }
+
+  const {data} = verifyTokenResponse || {};
+  const {appUserId} = data || {};
+
+  if (!appUserId) {
+    return res.status(500).json({
+      status: 'App user id is required',
+      code: 500
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The extraction of `exchangeToken` from `req.body` and its verification using `HumanIdService.verifyExchangeToken(exchangeToken)` is a good approach to validate the token. However, returning a 500 status code when the `appUserId` is not present might be misleading as it's usually used for server errors. A 400 (Bad Request) status code would be more appropriate here since it indicates that the client sent an invalid request.

```diff
-    return res.status(500).json({
+    return res.status(400).json({
       status: 'App user id is required',
-      code: 500
+      code: 400
     });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->